### PR TITLE
HPA docs

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1704956

### CODE REVIEW

**Comment by mburke5678** [comment]
@joelsmith  Do you have any info on how we are implementing HPA in 4.1? It appears that scaling for CPU is largely the same as in 3.11; but, scaling for memory utilization/custom metrics is different and doesn't seem to match upstream

**Comment by mburke5678** [comment]
@joelsmith When I used the 3.11 format to create the HPA object, I get:

```
The HorizontalPodAutoscaler "hpa-resource-metrics-cpu-1" is invalid: 
* spec.metrics[0].resource.target.type: Required value: must specify a metric target type
* spec.metrics[0].resource.target.type: Invalid value: "": must be either Utilization, Value, or AverageValue
* spec.metrics[0].resource.target.averageUtilization: Required value: must set either a target raw value or a target utilization

```
I got the object to build, but I would like to define `metric target type`, `Utilization`, `Value`, and `AverageValue` and what proper target values.

**Comment by mburke5678** [comment]
@joelsmith Please take a look. We are trying to close all PRs by the end of the week for GA. 

cc. @sjenning

**Comment by mburke5678** [comment]
@joelsmith PTAL Any input you can provide is greatly appreciate. Feel free to disregard typos and such. I want to get the technical details in place first.

File: modules/nodes-pods-autoscaling-about.adoc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+//
+// * nodes/nodes-pods-autoscaling-about.adoc
+
+[id='nodes-pods-autoscaling-about_{context}']
+= Understanding horizontal pod autoscalers
+
+You can create a horizontal pod autoscaler to specify the minimum and maximum number of pods you want to run, as well as the
+CPU utilization or memory utilization your pods should target.
+
+HPA, automatically scale the number of pods up and down for a specific Replication Controller or DeploymentConfig. You can specify the minimum and maximum number of pods
```
Comment (joelsmith) [inline modules/nodes-pods-autoscaling-about.adoc]: I think we should probably delete this paragraph. It restates the info from the preceding paragraph and starts off awkwardly. Or combine them if you want to keep the part about it updating the rc or dc with the replica value

**Review by joelsmith — COMMENTED** [review]
This is pretty close to being ready. Let me know if you have any questions about my suggestions. Thanks!

File: modules/nodes-pods-autoscaling-creating-cpu.adoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+[id='nodes-pods-autoscaling-creating-cpu_{context}']
+
+= Creating a horizontal pod autoscaler object for CPU utilization
+
+You can create a horizontal pod autoscaler to automatically scale pods when CPU usage exceeds a specified limit.
+
+.Prerequisites
+
+In order to use horizontal pod autoscalers, your cluster administrator must have properly configured cluster metrics.
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply modules/nodes-pods-autoscaling-creating-cpu.adoc]: Would output like this suggest metrics is installed?
```
oc describe PodMetrics openshift-kube-scheduler-ip-10-0-135-131.ec2.internal
Name:         openshift-kube-scheduler-ip-10-0-135-131.ec2.internal
Namespace:    openshift-kube-scheduler
Labels:       <none>
Annotations:  <none>
API Version:  metrics.k8s.io/v1beta1
Containers:
  Name:  wait-for-host-port
  Usage:
    Memory:  0
  Name:      scheduler
  Usage:
    Cpu:     8m
    Memory:  45440Ki
Kind:        PodMetrics
Metadata:
  Creation Timestamp:  2019-05-23T18:47:56Z
  Self Link:           /apis/metrics.k8s.io/v1beta1/namespaces/openshift-kube-scheduler/pods/openshift-kube-scheduler-ip-10-0-135-131.ec2.internal
Timestamp:             2019-05-23T18:47:56Z
Window:                1m0s
Events:                <none>
```

[the remark this answers is not in the document]
Comment (joelsmith) [reply modules/nodes-pods-autoscaling-creating-cpu.adoc]: Yes. If `Cpu` and `Memory` show up under `Usage`, then metrics are being gathered. If no service is providing metrics at all, the `oc` command will probably error out with an error like:
```
error: the server doesn't have a resource type "PodMetrics"
```

**Comment by mburke5678** [comment]
@xltian Please take a look at the PR. we are trying to close on all open issues by Thursday COD US EST (mid-day Friday Beijing). I apologize for getting this to you so late. Please let me know if meeting this deadline will be a problem.

**Comment by weinliu** [comment]
It LGTM

**Pull request merged**